libsolidity/ast/Types.cpp

```diff
@@ -1145,9 +1145,14 @@ TypePointer RationalNumberType::binaryOperatorResult(Token _operator, TypePointe
 
 				if (exp >= 0)
 					value = rational(numerator, denominator);
-				else
+				else {
 					// invert
-					value = rational(denominator, numerator);
+					// due to a bug in certain versions of boost the denominator cannot be negative
```

Comment:
This is rather confusing because you swap numerator and denominator. Would it be better to create a function that is used to construct the rational which then adjusts the signs accordingly?

Reply:
Yes, that's probably clearer, I moved it to a ``makeRational`` function.